drools-scenario-simulation/drools-scenario-simulation-backend/src/main/java/org/drools/scenariosimulation/backend/util/JsonUtils.java

```
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.scenariosimulation.backend.util;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Class used to provide JSON common utils
+ */
+public class JsonUtils {
+
+    private JsonUtils() {
+        // Not instantiable
+    }
+
+    /**
+     * Given a json in String format, it try to convert it in a <code>JsonNode</code>. In case of success, i.e.
+     * the given string is a valid json, it put the <code>JsonNode</code> in a <code>Optional</code>. An empty
+     * <code>Optional</code> is passed otherwise.
+     * @param json
+     * @return
+     */
+    public static Optional<JsonNode> convertFromStringToJSONNode(String json) {
+        if (json == null || json.isEmpty()) {
+           return Optional.empty();
+        }
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(json);
+            return Optional.of(jsonNode);
+        } catch (IOException e) {
+            return Optional.empty();
```

@yesamer 
Not sure about swallowing this exception that way: if we get here rawValue is expected to be a valid json, but if it throws exception, something bad has happened, IMO. 
"Given a json in String format, it try to convert it in a <code>JsonNode</code>. In case of success, i.e.
    the given string is a valid json"...: what would mean a "json in String format" but invalid json?

---

@gitgabrio This is a good point, I had the thought, but I didn't find a different way to validate a json. 
According to Jackson docs:

    

>  * @throws JsonParseException if underlying input contains invalid content
>      *    of type {@link JsonParser} supports (JSON for default case)
>      */
>     public JsonNode readTree(String content) throws IOException {
>         return _readTreeAndClose(_jsonFactory.createParser(content));
>     }

Then, I can improve it managing `JsonParseException` and `IOException` in different ways: in the first case, it's a malformed content i.e. not a JSON, in the second case It could be any other IOException, I can re-throw it with an other exception (which one?)

---

@yesamer 
`JsonParseException` extends `IOException`.
If I've understood correctly, `ObjectMapper.readTree(String content)` may throw one or the other (it declares only the latter, being the ancestor).
So, I think both of them should be catch and managed in different way (if the json is invalid, I definitely would not hide that - for other underlying IO exceptions is debatable).
This could also be tested, I think.

---

@gitgabrio no, a `JsonParseException` is thrown if the input is not a valid json. In this case, I return an empty `Optional` which means: the given string can't be parsed as json and then it's not a json. In all other cases, when a generic `IOException` is thrown, then I re-throw an `IllegalArgumentException`

---

@yesamer 
I thought that method would be invoked *only* to parse an (expected) json string. If this is true, then IMO `JsonParseException` should not be hided behind an `Optional.empty()`, because it means something unexpected is happening. If - instead - this method is invoked on any possible String to verify if it is a valid json, then the javadoc should somehow clarify that

---

@gitgabrio 
> If - instead - this method is invoked on any possible String to verify if it is a valid json,
This is the correct case, I'll updated the javadoc to clarify it.